hs/src/Reach/Eval/Core.hs

```diff
@@ -2387,15 +2426,20 @@ evalPrim p sargs =
           DLE_TokenNew at tns
       tokdv <- doInternalLog_ Nothing tokdv_
       st <- readSt id
+      let existingToks = st_toks st
+      tokIsUniq <- tokIsUnique (map DLA_Var existingToks) $ DLA_Var tokdv
```

Comment:
When creating a token, assume that it's unique from all the existing tokens. This is needed for verifying that the tokens in a pay amount are unique

Reply:
This is a valid assumption too :)